src/main/java/se/sundsvall/precheck/integration/partyassets/PartyAssetsConfiguration.java

```diff
@@ -0,0 +1,57 @@
+package se.sundsvall.precheck.integration.partyassets;
+
+import feign.Request;
+import feign.codec.ErrorDecoder;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.openfeign.FeignBuilderCustomizer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.core.AuthorizationGrantType;
+import se.sundsvall.dept44.configuration.feign.FeignConfiguration;
+import se.sundsvall.dept44.configuration.feign.FeignMultiCustomizer;
+import se.sundsvall.dept44.configuration.feign.decoder.ProblemErrorDecoder;
+
+import java.util.concurrent.TimeUnit;
+
+@Import(FeignConfiguration.class)
+@EnableConfigurationProperties(PartyAssetsProperties.class)
+public class PartyAssetsConfiguration {
+
+    private final PartyAssetsProperties properties;
+
+    PartyAssetsConfiguration(final PartyAssetsProperties properties) {
+        this.properties = properties;
+    }
+
+    @Bean
+    FeignBuilderCustomizer partyAssetsCustomizer() {
+        return FeignMultiCustomizer.create()
+                .withErrorDecoder(errorDecoder())
+                .withRequestOptions(requestOptions())
+                .withRetryableOAuth2InterceptorForClientRegistration(clientRegistration())
+                .composeCustomizersToOne();
+    }
+
+    Request.Options requestOptions() {
+        return new Request.Options(
+                properties.getConnectTimeout().toMillis(), TimeUnit.MILLISECONDS,
+                properties.getReadTimeout().toMillis(), TimeUnit.MILLISECONDS,
+                false);
+
+    }
+
+    ClientRegistration clientRegistration() {
+        return ClientRegistration
+                .withRegistrationId(PartyAssetsIntegration.INTEGRATION_NAME)
+                .tokenUri(properties.getOAuth2().getTokenUri())
```

Comment:
Not needed if you name your properties according to documentation (have a look at some other services)

Reply:
Not really, but this code is straight up taken from the ``'api-service-case-status'`` service's [``'CitizenIntegrationConfiguration.java'``](https://github.com/Sundsvallskommun/api-service-case-status/blob/main/src/main/java/se/sundsvall/casestatus/integration/citizen/CitizenIntegrationConfiguration.java) at lines 44 to 52.

Should I redo it according to another service, and if so, which one do you recommend?